apps/hash-frontend/src/pages/account/integrations/select-linear-teams-table.tsx

```diff
@@ -0,0 +1,332 @@
+        )
+        .map(
+          ({ workspaceEntity }) => workspaceEntity.metadata.recordId.entityId,
+        ),
+    })),
+  });
+
+export const mapLinearOrganizationToSyncWithWorkspacesInputVariable = (params: {
+  linearOrganization: LinearOrganizationTeamsWithWorkspaces;
+  possibleWorkspaces: (Org | MinimalUser)[];
+}): SyncWithWorkspace[] =>
+  params.possibleWorkspaces
+    .filter(({ entityRecordId: { entityId } }) =>
+      params.linearOrganization.teams.some(({ workspaceEntityIds }) =>
+        workspaceEntityIds.includes(entityId),
+      ),
+    )
+    .map(({ entityRecordId: { entityId: workspaceEntityId } }) => {
+      const linearTeamIds = params.linearOrganization.teams
+        .filter(({ workspaceEntityIds }) =>
+          workspaceEntityIds.includes(workspaceEntityId),
+        )
+        .map(({ id }) => id);
+
+      /** @todo: allow the user to decide whether to sync future teams */
+      const syncAllTeams =
+        params.linearOrganization.teams.length === linearTeamIds.length;
+
+      return {
+        workspaceEntityId,
+        linearTeamIds: syncAllTeams ? [] : linearTeamIds,
+      };
+    });
+
+export const SelectLinearTeamsTable: FunctionComponent<{
+  linearOrganizations: LinearOrganizationTeamsWithWorkspaces[];
+  setLinearOrganizations: Dispatch<
+    SetStateAction<LinearOrganizationTeamsWithWorkspaces[]>
+  >;
+}> = ({ linearOrganizations, setLinearOrganizations }) => {
+  const { authenticatedUser } = useAuthenticatedUser();
+
+  const possibleWorkspaces = [authenticatedUser, ...authenticatedUser.memberOf];
+
+  return (
+    <Table sx={{ minWidth: 650 }} aria-label="simple table">
+      <TableHead>
+        <TableRow>
+          <TableCell>
+            <strong>Type</strong> in Linear
+          </TableCell>
+          <TableCell>
+            <strong>Name</strong> in Linear
+          </TableCell>
+          <TableCell>HASH workspace(s) to sync with</TableCell>
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        {linearOrganizations.map(
+          ({ id: linearOrgId, name: orgName, teams }) => (
+            <Fragment key={linearOrgId}>
+              <TableRow>
+                <TableCell>Workspace</TableCell>
+                <TableCell>{orgName}</TableCell>
+                <TableCell>
+                  <SelectWorkspaces
+                    selectedWorkspaceEntityIds={possibleWorkspaces
+                      .map(({ entityRecordId: { entityId } }) => entityId)
+                      .filter(
+                        (entityId) =>
+                          teams.length ===
+                          teams.filter(({ workspaceEntityIds }) =>
+                            workspaceEntityIds.includes(entityId),
+                          ).length,
+                      )}
+                    possibleWorkspaces={possibleWorkspaces}
+                    setSelectedWorkspaceEntityIds={(entityIds) =>
+                      setLinearOrganizations((prev) => {
```

Comment:
probably not much of a difference but if we ended up syncing lots of orgs this would be a good candidate for moving to a callback somewhere higher up to avoid creating this function on every render as many times as there are Linear orgs synced

Reply:
That's a good point, I've moved them into callbacks in https://github.com/hashintel/hash/pull/2826/commits/680c4c03fce014db7a03fb02dc066248320e562c